spring-cloud-netflix-eureka-server/src/main/java/org/springframework/cloud/netflix/eureka/server/EurekaServerBootstrap.java

```diff
@@ -0,0 +1,109 @@
+import com.netflix.eureka.registry.PeerAwareInstanceRegistry;
+import com.netflix.eureka.util.EurekaMonitors;
+import com.thoughtworks.xstream.XStream;
+
+/**
+ * @author Spencer Gibb
+ */
+@CommonsLog
+public class EurekaServerBootstrap extends EurekaBootStrap {
+	private static final String TEST = "test";
+
+	private static final String ARCHAIUS_DEPLOYMENT_ENVIRONMENT = "archaius.deployment.environment";
+
+	private static final String EUREKA_ENVIRONMENT = "eureka.environment";
+
+	private static final String DEFAULT = "default";
+
+	private static final String ARCHAIUS_DEPLOYMENT_DATACENTER = "archaius.deployment.datacenter";
+
+	private static final String EUREKA_DATACENTER = "eureka.datacenter";
+
+	private EurekaServerConfig eurekaServerConfig;
+
+	private ApplicationInfoManager applicationInfoManager;
+
+	private EurekaClientConfig eurekaClientConfig;
+
+	private PeerAwareInstanceRegistry registry;
+
+	public EurekaServerBootstrap(ApplicationInfoManager applicationInfoManager, EurekaClientConfig eurekaClientConfig, EurekaServerConfig eurekaServerConfig, PeerAwareInstanceRegistry registry, EurekaServerContext serverContext) {
+		this.applicationInfoManager = applicationInfoManager;
+		this.eurekaClientConfig = eurekaClientConfig;
+		this.eurekaServerConfig = eurekaServerConfig;
+		this.registry = registry;
+		this.serverContext = serverContext;
+	}
+
+	@Override
+	protected void initEurekaEnvironment() throws Exception {
+		log.info("Setting the eureka configuration..");
+
+		String dataCenter = ConfigurationManager.getConfigInstance().getString(EUREKA_DATACENTER);
+		if (dataCenter == null) {
+			log.info("Eureka data center value eureka.datacenter is not set, defaulting to default");
+			ConfigurationManager.getConfigInstance().setProperty(ARCHAIUS_DEPLOYMENT_DATACENTER, DEFAULT);
+		} else {
+			ConfigurationManager.getConfigInstance().setProperty(ARCHAIUS_DEPLOYMENT_DATACENTER, dataCenter);
+		}
+		String environment = ConfigurationManager.getConfigInstance().getString(EUREKA_ENVIRONMENT);
+		if (environment == null) {
+			ConfigurationManager.getConfigInstance().setProperty(ARCHAIUS_DEPLOYMENT_ENVIRONMENT, TEST);
+			log.info("Eureka environment value eureka.environment is not set, defaulting to test");
+		}
+	}
+
+	@Override
+	protected void initEurekaServerContext() throws Exception {
+		// For backward compatibility
+		JsonXStream.getInstance().registerConverter(new V1AwareInstanceInfoConverter(), XStream.PRIORITY_VERY_HIGH);
+		XmlXStream.getInstance().registerConverter(new V1AwareInstanceInfoConverter(), XStream.PRIORITY_VERY_HIGH);
+
+		if (isAws(applicationInfoManager.getInfo())) {
+			awsBinder = new AwsBinderDelegate(eurekaServerConfig, eurekaClientConfig, registry, applicationInfoManager);
+			awsBinder.start();
+		}
+
+		EurekaServerContextHolder.initialize(serverContext);
+
+		log.info("Initialized server context");
+
+		// Copy registry from neighboring eureka node
+		int registryCount = registry.syncUp();
```

Comment:
This is the line that causes the 2.5 min delay. It also causes the errors mentioned.